src/main/java/com/iexec/core/configuration/PurgeConfiguration.java

```diff
@@ -0,0 +1,16 @@
+package com.iexec.core.configuration;
+
+import com.iexec.common.utils.purge.PurgeService;
+import com.iexec.common.utils.purge.Purgeable;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.List;
+
+@Configuration
+public class PurgeConfiguration {
+    @Bean
+    PurgeService purgeService(List<Purgeable> purgeableServices) {
+        return new PurgeService(purgeableServices);
+    }
```

Comment:
Or is spring able to find all beans implementing `Purgeable` and to inject them here ?

Reply:
Spring is able to find and inject all beans implementing `Purgeable`. That means `SmsClientProvider` will be injected there - as it is declared as a bean in [PurgeConfiguration](https://github.com/iExecBlockchainComputing/iexec-core/blob/feature/gramine-purge-maps/src/main/java/com/iexec/core/configuration/PurgeConfiguration.java) -, and other beans could be added later if needed.

Reply:
There are several implementation points I would like to ask about
* This implementation assumes there will always exist at least an existing bean implementing `Purgeable`. Is it true or should we handle the situation where the list is empty ?
* Are we sure the `PurgeService` instance will be created **after** all beans implementing `Purgeable` have been instantiated ? What will happen if a bean is created after ?
* We don't have a list of injected beans. Maybe we could just add log events in constructor or in a `@PostConstruct` ?
* We have not spoken about application shutdown. What should happen when beans are interrupted and closed ? How do we stop and close beans implementing `Purgeable` ?

Reply:
Let's try and reply to all these interrogations! :grin: 
- Please see [Spring documentation](https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#beans-autowired-annotation). There's a special case for injected constructors: "Injected constructor and factory method arguments are a special case [...] multi-element injection points (arrays, collections, maps) resolving to empty instances if no matching beans are available.". So this would resolve to an empty list there. No worries! Documentation added in https://github.com/iExecBlockchainComputing/iexec-core/pull/522/commits/1a9d33b33b87c6e691a6ae79135869dfebd43986.
- I don't find anything on that point in Spring documentation. The only relevant point I can see is that we should trust Spring on dependency resolution.
- Let's add it in `PurgeService` constructor, thanks! See https://github.com/iExecBlockchainComputing/iexec-common/pull/355/commits/d6f39f680e94442948dc0a6ff5ccaa3795fbb144.
- Good point. We should purge all task-related data in those services. Please see the following commits:
  - https://github.com/iExecBlockchainComputing/iexec-common/pull/355/commits/6e522f76582d4ec124d650b51d800871e2f3b44a
  - https://github.com/iExecBlockchainComputing/iexec-sms/pull/121/commits/0a2fcc500ce1c2aa8928cf77ae0fb5ed23e76d2b
  - https://github.com/iExecBlockchainComputing/iexec-worker/pull/471/commits/e42e5c4a5fc18ad050e4c20dd6884dbd17a1058e